database.go

```diff
@@ -0,0 +1,1235 @@
+
+type databaseNode struct {
+	client *Client
+}
+
+type databaseTask struct {
+	client *Client
+}
+
+type databaseConfiguration struct {
+	client *Client
+}
+
+type databaseBackup struct {
+	client *Client
+}
+
+type databaseSchedule struct {
+	client *Client
+}
+
+type databaseAutoscaling struct {
+	client *Client
+}
+
+type databaseFirewall struct {
+	client *Client
+}
+
+type databaseEngine struct {
+	client *Client
+}
+
+// Instance Resource Path
+func (ins *databaseInstance) resourcePath(instanceID string) string {
+	return databaseInstancesResourcePath + "/" + instanceID
+}
+
+// Instance Action Path
+func (ins *databaseInstance) resourceActionPath(instanceID string) string {
+	return databaseInstancesResourcePath + "/" + instanceID + "/action"
+}
+
+// Node Resource Path
+func (n *databaseNode) resourcePath(nodeID string) string {
+	return databaseNodesResourcePath + "/" + nodeID
+}
+
+// Node Resource Action Path
+func (n *databaseNode) resourceActionPath(nodeID string) string {
+	return databaseNodesResourcePath + "/" + nodeID + "/action"
+}
+
+// Configuration Resource Path
+func (cfg *databaseConfiguration) resourcePath(cfgID string) string {
+	return databaseConfigurationsResourcePath + "/" + cfgID
+}
+
+// Configuration Resource Action Path
+func (cfg *databaseConfiguration) resourceActionPath(nodeID string, cfgID string) string {
+	return databaseNodesResourcePath + "/" + nodeID + "/configuration/" + cfgID
+}
+
+// Backup Resource Path
+func (bk *databaseBackup) resourcePath(backupID string) string {
+	return databaseBackupsResourcePath + "/" + backupID
+}
+
+// Backup Resource Create Path
+func (bk *databaseBackup) resourceCreatePath(resourceType string, resourceID string) string {
+	return "/" + resourceType + "/" + resourceID + databaseBackupsResourcePath
+}
+
+// Schedule Resource List Path
+func (sc *databaseSchedule) resourcePath(scheduleID string) string {
+	return databaseSchedulesResourcePath + "/" + scheduleID
+}
+
+// Schedule List In Resource Path
+func (sc *databaseSchedule) resourceOtherPath(resource string, resourceID string) string {
+	return "/" + resource + "/" + resourceID + databaseSchedulesResourcePath
+}
+
+// Schedule Resource Backup Path
+func (sc *databaseSchedule) resourceBackupListPath(scheduleID string) string {
+	return databaseSchedulesResourcePath + "/" + scheduleID + databaseBackupsResourcePath
+}
+
+// Au Resource Path
+func (au *databaseAutoscaling) resourcePath(resourceID string) string {
+	return databaseInstancesResourcePath + "/" + resourceID + databaseAutoscalingResourcePath
+}
+
+// fw Resource Path
+func (fw *databaseFirewall) resourcePath(nodeID string) string {
+	return databaseNodesResourcePath + "/" + nodeID + databaseTrustedSourcesResourcePath
+}
+
+// ta Resource Path
+func (ta *databaseTask) resourcePath(taskID string) string {
+	return databaseTasksResourcePath + "/" + taskID + "status"
+}
+
+// DatabaseListOption contains option when list resource
+type DatabaseListOption struct {
```

Comment:
what difference between this and other list options?

Reply:
why we decide make more list options? Can't re-use, right it's?

Reply:
Because cloud database have a difference field `name` to query when list so we need to create a new one struct to define. I will update it to inherit from old struct.